libs/vm-modules/tests/unit/ml/ml_model_tests.cpp

```diff
@@ -1000,6 +1011,25 @@ TEST_F(VMModelTests, model_sequential_avgpool2d)
   ASSERT_TRUE(toolkit.Run(nullptr, ChargeAmount{0}));
 }
 
+TEST_F(VMModelTests, model_sequential_embeddings)
+{
+  static char const *sequential_model_src = R"(
+        function main()
+          var x = Tensor();
+          var str_vals = "1.5; 2.1; 3.1; 4.2; 5.1; 6.; 7.1; 8.1; 9.0;";
+          x.fromString(str_vals);
+          var model = Model("sequential");
+          model.addExperimental("input", x.shape());
+          model.addExperimental("embeddings", 10u64, 4u64, true);
```

Comment:
looks a bit confusing: aren't the inputs meant to be row indices? what actually happens internally when you run this?

Reply:
To be precise, static cast is happening
auto embedding_view = this->data_->View(static_cast<SizeType>(*input_it));
But you're absolutely right, I'll change the input to be less confusing